src/Geometries/VolumePrimitives/HexagonalPrism.jl

```diff
@@ -0,0 +1,144 @@
+            push!(pts_bottom_inner, pt_bottom_inner)
+        end
+
+        #create Linesegments connecting the vertices
+        lines = LineSegment{T, 3, :cartesian}[]
+        N = length(pts_top_outer)
+        for i in 1:N
+            push!(lines, LineSegment(pts_top_outer[i%N+1], pts_top_outer[i])) #top outer hexagon
+            push!(lines, LineSegment(pts_bottom_outer[i%N+1], pts_bottom_outer[i])) #bottom outer hexagon
+            push!(lines, LineSegment(pts_bottom_outer[i], pts_top_outer[i])) #lines connecting outer hexagons
+            if hp.rInner > 0
+                push!(lines, LineSegment(pts_bottom_inner[i%N+1], pts_bottom_inner[i])) #bottom inner hexagon
+                push!(lines, LineSegment(pts_top_inner[i%N+1], pts_top_inner[i])) #top inner hexagon
+                push!(lines, LineSegment(pts_bottom_inner[i], pts_top_inner[i])) #lines connecting inner hexagons
+                push!(lines, LineSegment(pts_top_outer[i], pts_top_inner[i])) #lines connecting hexagons
+                push!(lines, LineSegment(pts_bottom_outer[i], pts_bottom_inner[i])) #lines connecting hexagons
+            end
+        end
+        lines
+    end
+end
+
+# For proper grid creation we also need the function get_important_points:
+# Radial
+function get_important_points(hp::HexagonalPrism{T}, ::Val{:r})::Vector{T} where {T <: SSDFloat}
+    return geom_round.(T[-hp.rOuter, -sqrt(3)/2*hp.rOuter, -hp.rInner, -sqrt(3)/2*hp.rInner, 0., sqrt(3)/2*hp.rInner, hp.rInner, sqrt(3)/2*hp.rOuter, hp.rOuter] .+ sqrt(hp.translate.x^2 + hp.translate.y^2))
+end
+
+# polar angle
+function get_important_points(hp::HexagonalPrism{T}, ::Val{:φ})::Vector{T} where {T <: SSDFloat}
+    return T[]
+end
+# Z
+function get_important_points(hp::HexagonalPrism{T}, ::Val{:z})::Vector{T} where {T <: SSDFloat}
+    return geom_round.(T[-hp.h/2, hp.h/2] .+ hp.translate.z)
+end
+# X
+function get_important_points(hp::HexagonalPrism{T}, ::Val{:x})::Vector{T} where {T <: SSDFloat}
+    return geom_round.(T[-hp.rOuter, -sqrt(3)/2*hp.rOuter, -hp.rInner, -sqrt(3)/2*hp.rInner, 0., sqrt(3)/2*hp.rInner, hp.rInner, sqrt(3)/2*hp.rOuter, hp.rOuter] .+ hp.translate.x)
+end
+# Y
+function get_important_points(hp::HexagonalPrism{T}, ::Val{:y})::Vector{T} where {T <: SSDFloat}
+    return geom_round.(T[-hp.rOuter, -sqrt(3)/2*hp.rOuter, -hp.rInner, -sqrt(3)/2*hp.rInner, 0., sqrt(3)/2*hp.rInner, hp.rInner, sqrt(3)/2*hp.rOuter, hp.rOuter] .+ hp.translate.y)
+end
+
+#and a sample function to paint the primitive on the grid (necessary if the object is small)
+function sample(hp::HexagonalPrism{T}, stepsize::Vector{T})  where T
+    samples = CartesianPoint{T}[]
+    for x in -hp.rOuter : stepsize[1] : hp.rOuter
+        for y in -hp.rOuter : stepsize[2]: hp.rOuter
+            for z in -hp.h/2 : stepsize[3] : hp.h/2
+                p = CartesianPoint{T}(x, y, z)
+                if in(samples, p)
+                    push!(samples, p)
```

Comment:
@leenderthayen: I don't understand this two lines of code. 
This `if`-statement is always false (`samples` cannot be in `p`).

If you are trying to only `push!`, if `p` is in `samples`, you could use:
```julia
if in(p,samples)
    push!(samples,p)
end
```

However, as `samples` is empty at the beginning, this will also never happen.

Did you intend to write:
```julia
if !in(p,samples)
    push!(samples,p)
end
```
?

Reply:
Hmm, actually I believe it should be 
```julia
if p in hp push!(samples, p) end
```
?

Reply:
PR: #86